### PR TITLE
Change to use es6

### LINKED ISSUE #57: Change to es6

### PR DESCRIPTION
**Pull request by julyytran**
This PR updates all of the classes to use ES6 class/constructor style, and changes all `var` to `let`. Reviewers may want to start looking at the simpler classes (sushi, trash, heart, cat) and then to the more complex ones (scoreboard, game). This should be manually tested by playing the game and ensuring all the functionality still works. Are there any more es6-y things to do? @rrgayhart @cheljoh @julsfelic

Closes #57

### CODE REVIEW

File: lib/background.js [hunk 1]
```diff
@@ -1,45 +1,47 @@
...
-Background.prototype.randomStarsImage = function(){
-  this.getStars();
+  randomStarsImage() {
+    this.getStars();
 
-  var img = document.createElement("img");
-  img.src = this.starfield.toDataURL();
-  return img;
-};
+    var img = document.createElement("img");
```
Comment (julsfelic) [inline lib/background.js]: I suggest using `let` and avoiding the use of `var` due to unexpected behavior of `var`

File: lib/background.js [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
-    var coordinates = this.getStarCoordinates();
-    var x = coordinates.x;
-    var y = coordinates.y;
-    var radius = coordinates.radius;
-    this.starfieldCtx.arc(x, y, radius, 0, Math.PI * 2,false);
-    this.starfieldCtx.closePath();
+  getStars() {
+    this.starfieldCtx.beginPath();
+    for(var n = 0 ; n < 100; n++){
+      var coordinates = this.getStarCoordinates();
```
Comment (julsfelic) [inline lib/background.js]: Same as above.

File: lib/background.js [hunk 3]
```diff
@@ -1,45 +1,47 @@
...
-};
 
-Background.prototype.getStarCoordinates = function(){
-  var x = parseInt(Math.random() * this.canvas.width);
-  var y = parseInt(Math.random() * this.canvas.height);
-  var radius = Math.random() * 3;
-  return { x: x, y: y, radius: radius };
-};
+  getStarCoordinates() {
+    var x = parseInt(Math.random() * this.canvas.width);
```
Comment (julsfelic) [inline lib/background.js]: See above

File: lib/draw.js [hunk 4]
```diff
@@ -14,41 +14,43 @@ var nyanFrames = [nyan1, nyan2, nyan3, nyan4, nyan5, nyan6, nyan7, nyan8, nyan9,
...
-Draw.prototype.drawCat = function(cat){
-  var now = Date.now();
-  var elapsed = (now - lastFrameTime);
-  var frame = nyanFrames[frameCount];
-  this.checkNewFrame(elapsed, now);
-  cat.context.drawImage(frame, cat.x, cat.y);
-  this.resetFrameCount();
-};
+  drawCat(cat) {
+    var now = Date.now();
```
Comment (julsfelic) [inline lib/draw.js]: See above

File: lib/game.js [hunk 5]
```diff
@@ -8,136 +8,138 @@ const Scoreboard = require('./scoreboard');
 var draw = new Draw();
```
Comment (julsfelic) [inline lib/game.js]: See above

File: lib/game.js [hunk 6]
```diff
@@ -8,136 +8,138 @@ const Scoreboard = require('./scoreboard');
...
 var helpers = new Helpers();
 var scoreboard = new Scoreboard();
```
Comment (julsfelic) [inline lib/game.js]: See above

**Comment by julsfelic** [comment]
Everything is looking ES6ified! Once the `vars` are changed to `lets` it'll be good to merge!

**Comment by julyytran** [comment]
Thanks, just changed them all!

File: lib/background.js [hunk 7]
```diff
@@ -1,45 +1,47 @@
...
-  this.starfield = options.starfield;
-  this.starfieldCtx = options.starfieldCtx;
-  this.canvas = options.canvas;
-}
+class Background {
+  constructor(options) {
+    this.starfield = options.starfield;
+    this.starfieldCtx = options.starfieldCtx;
+    this.canvas = options.canvas;
+  }
```
Comment (cheljoh) [inline lib/background.js:6]: yeah class!!!

**Comment by cheljoh** [comment]
Nice work July! Let me know when Babel is up and I will merge.

**Comment by julyytran** [comment]
Added config for bable, could you check it works with IE?

**Comment by cheljoh** [comment]
I got it running locally, going to merge!

**Comment by rrgayhart** [comment]
Noice work! Copying and pasting the response I left on Brian's PR:

> Keep an eye on matching the requirements for assignments, even if they seem weird. For example, this PR didn't use the PR template as requested - not a big deal because you provide good detail, but ignoring that requirement would look really bad on a code challenge for a company.

File: .babelrc [hunk 8]
```diff
@@ -0,0 +1 @@
+{ "presets": ["es2015"] }
```
Comment (rrgayhart) [inline .babelrc:1]: 👍

**Pull request merged**